### PR TITLE
Run codespell on pull requests

### PR DESCRIPTION
**Pull request by pemensik**
Related to issue #167

### CODE REVIEW

**Comment by evverx** [comment]
FWIW the utf8 errors can be fixed by applying https://github.com/lathiat/avahi/pull/425/commits/bad042c006221b8cff38d3079b2ea745d797a626. clang doesn't like it either.

**Comment by pemensik** [comment]
> FWIW the utf8 errors can be fixed by applying [bad042c](https://github.com/lathiat/avahi/commit/bad042c006221b8cff38d3079b2ea745d797a626). clang doesn't like it either.

Is that commit part of any PR? I think we want similar change. I would use \xfc hex encoding myself, but we do not want to rely on file encoded in latin-1 anymore. Especially if it tries garbage more or less.

**Comment by evverx** [comment]
> Is that commit part of any PR?

As far as I can remember it was included in a PR where `-Werror` was passed to clang. There were (and still are) so many errors that I closed it but GitHub keeps those commits.

**Comment by pemensik** [comment]
I think it is ready

File: .github/workflows/codespell.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+permissions:
+  contents: read
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-${{ matrix.language }}-${{ github.ref }}
```
Comment (evverx) [inline .github/workflows/codespell.yml:20]: Looks like `${{ matrix.language }}` came from the codeql workflow and can be removed.

**Review by evverx — COMMENTED** [review]
I think it would be great to run this on PRs but I'm not sure the code should change. Would it be possible to ignore `fo`, `te`, `etxt` and `calback` instead for the time being?

**Comment by evverx** [comment]
The commits related to the utf8 tests were merged in https://github.com/lathiat/avahi/pull/481

**Pull request closed without merging**